opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py

```diff
@@ -390,8 +421,55 @@ def _get_from_env_or_default() -> Sampler:
     return _KNOWN_SAMPLERS[trace_sampler]
 
 
+def _get_from_env_or_default() -> Sampler:
+    trace_sampler_name = os.getenv(
+        OTEL_TRACES_SAMPLER, "parentbased_always_on"
+    ).lower()
+
+    if trace_sampler_name in _KNOWN_SAMPLERS:
+        if trace_sampler_name in ("traceidratio", "parentbased_traceidratio"):
+            try:
+                rate = float(os.getenv(OTEL_TRACES_SAMPLER_ARG))
+            except ValueError:
+                _logger.warning(
+                    "Could not convert TRACES_SAMPLER_ARG to float."
+                )
+                rate = 1.0
+            return _KNOWN_SAMPLERS[trace_sampler_name](rate)
+        return _KNOWN_SAMPLERS[trace_sampler_name]
+    else:
+        try:
+            trace_sampler_factory = _import_sampler_factory(trace_sampler_name)
+            sampler_arg = os.getenv(OTEL_TRACES_SAMPLER_ARG, "")
+            trace_sampler = trace_sampler_factory(sampler_arg)
+            _logger.warning("JEREVOSS: trace_sampler: %s" % trace_sampler)
+            if not issubclass(type(trace_sampler), Sampler):
```

Comment:
They're different use cases. issubclass is for classes not objects of said class. Since the entry point is just a function, we have to check the resulting object's type, not the entry point's class

Reply:
Done